pkgs/tools/misc/cowsay/default.nix

```diff
@@ -1,34 +1,44 @@
-{ lib, stdenv, fetchurl, perl }:
+{ lib, stdenv, perl, installShellFiles, fetchFromGitHub }:
 
 stdenv.mkDerivation rec {
-  version = "3.03+dfsg2";
   pname = "cowsay";
+  version = "3.04";
 
-  src = fetchurl {
-    url = "http://http.debian.net/debian/pool/main/c/cowsay/cowsay_${version}.orig.tar.gz";
-    sha256 = "0ghqnkp8njc3wyqx4mlg0qv0v0pc996x2nbyhqhz66bbgmf9d29v";
+  src = fetchFromGitHub {
+    owner = "tnalpgge";
+    repo = "rank-amateur-cowsay";
+    rev = "cowsay-${version}";
+    sha256 = "sha256-9jCaQ6Um6Nl9j0/urrMCRcsGeubRN3VWD3jDM/AshRg=";
   };
 
   buildInputs = [ perl ];
 
-  postBuild = ''
+  nativeBuildInputs = [ installShellFiles ];
+
+  # overriding buildPhase because we don't want to use the install.sh script
+  buildPhase = ''
+    runHook preBuild;
     substituteInPlace cowsay --replace "%BANGPERL%" "!${perl}/bin/perl" \
       --replace "%PREFIX%" "$out"
+    runHook postBuild;
   '';
 
   installPhase = ''
-    mkdir -p $out/{bin,man/man1,share/cows}
-    install -m755 cowsay $out/bin/cowsay
-    ln -s cowsay $out/bin/cowthink
-    install -m644 cowsay.1 $out/man/man1/cowsay.1
-    ln -s cowsay.1 $out/man/man1/cowthink.1
-    install -m644 cows/* -t $out/share/cows/
+    runHook preInstall
+    install -Dm755 cowsay $out/bin/cowsay
+    ln -s $out/bin/cowsay $out/bin/cowthink
+    ln -s cowsay.1 cowthink.1
+    installManPage cowsay.1 cowthink.1
```

Comment:
This is the the wrong way round. We should create a symlink in `$out` to reduce closure size.

Reply:
I can't really get that to work, `installManPage` automatically seems to install the man page in the "man" part, which I don't have a reliable way of accessing (i think?)

If I `install` the man page manually, and then make a symlink, it fails with this error:
```
moving /nix/store/qgvjz29x3y9qxjyd92wlxn3akj7niwxq-cowsay-3.04/man to /nix/store/qgvjz29x3y9qxjyd92wlxn3akj7niwxq-cowsay-3.04/share/man
Moving /nix/store/qgvjz29x3y9qxjyd92wlxn3akj7niwxq-cowsay-3.04/share/man to /nix/store/swj9d21ma3a5jp4gcxf3qn0wd6pmrhaf-cowsay-3.04-man/share/man
rmdir: failed to remove '/nix/store/qgvjz29x3y9qxjyd92wlxn3akj7niwxq-cowsay-3.04/share': Directory not empty
```

Reply:
If you want to keep multiple outputs (I don't see why) you have to use `$man` instead of `$out`.

Reply:
`ln -s cowsay.1 $man/share/man/man1/cowthink.1` should work